### PR TITLE
test: use popen3 to run shell commands

### PR DESCRIPTION
**Pull request by cpanato**

### CODE REVIEW

File: tests/rspec/lib/metal_support.rb [hunk 1]
```diff
@@ -128,4 +156,16 @@ def self.env_variables_setup
...
   end
+
+  def self.execute_command(cmd)
+    Open3.popen3(cmd) do |_stdin, stdout, _stderr, wait_thr|
+      exit_status = wait_thr.value
+      unless exit_status.success?
+        while (line = stdout.gets)
+          puts line
+        end
+        raise "Comand execution FAILED !!! #{cmd}"
```
Comment (mxinden) [inline tests/rspec/lib/metal_support.rb]: s/comand/command

Comment (cpanato) [reply tests/rspec/lib/metal_support.rb]: thanks!

**Comment by cpanato** [comment]
@mxinden you can have another look when you have time 😄

File: tests/rspec/lib/metal_support.rb [hunk 2]
```diff
@@ -24,13 +25,13 @@
...
-    `sudo mv /tmp/terraform /usr/local/bin/`
-    `sudo chmod +x /usr/local/bin/terraform`
-    `(cd #{root}/ && rm -rf matchbox && git clone https://github.com/coreos/matchbox)`
-    `(cd #{root}/matchbox && git checkout #{MATCHBOX_VERSION})`
+    execute_command("sudo curl -L -o /usr/local/bin/kubectl #{KUBECTL_URL}")
+    execute_command('sudo chmod +x /usr/local/bin/kubectl')
+    execute_command("curl #{TERRAFORM_URL} | funzip > /tmp/terraform")
+    execute_command('sudo mv /tmp/terraform /usr/local/bin/')
+    execute_command('sudo chmod +x /usr/local/bin/terraform')
+    execute_command("(cd #{root}/ && rm -rf matchbox && git clone https://github.com/coreos/matchbox)")
```
Comment (mxinden) [inline tests/rspec/lib/metal_support.rb]: Are the brackets inside the `"` needed?

File: tests/rspec/lib/metal_support.rb [hunk 3]
```diff
@@ -128,4 +156,16 @@ def self.env_variables_setup
       'ASSETS_DIR' => '/tmp/matchbox/assets'
     }
   end
+
+  def self.execute_command(cmd)
+    Open3.popen3(cmd) do |_stdin, stdout, _stderr, wait_thr|
+      exit_status = wait_thr.value
+      unless exit_status.success?
```
Comment (mxinden) [inline tests/rspec/lib/metal_support.rb]: If the exit status is not `0`, why not print out `stderr` as well?

File: tests/rspec/lib/metal_support.rb (hunk 2 above)
Comment (cpanato) [reply tests/rspec/lib/metal_support.rb]: maybe not

File: tests/rspec/lib/metal_support.rb (hunk 3 above)
Comment (cpanato) [reply tests/rspec/lib/metal_support.rb]: will remove

Comment (mxinden) [reply tests/rspec/lib/metal_support.rb]: @cpanato "Remove"? I though about adding something like:

```Ruby
while (line = stderr.gets)
         puts line
end
```

What do you think? Am I missing something here?

Comment (cpanato) [reply tests/rspec/lib/metal_support.rb]: ahh, got it, ok

File: tests/rspec/lib/metal_support.rb (hunk 2 above)
Comment (cpanato) [reply tests/rspec/lib/metal_support.rb]: done

File: tests/rspec/lib/metal_support.rb (hunk 3 above)
Comment (cpanato) [reply tests/rspec/lib/metal_support.rb]: done

**Pull request merged**